### PR TITLE
[Spark] DynamoDBCommitOwner: add logging, get dynamic confs from sparkSession

### PR DESCRIPTION
**Pull request by dhruvarya-db**
#### Which Delta project/connector is this regarding?

- [X] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [ ] Other (fill in here)

## Description

Updates DynamoDBCommitOwner:

 - Added logging around table creation flow
 - Get wcu, rcu, and awsCredentialsProvider from SparkSession
 - Return -1 as the table version if registerTable has already been called but no actual commits have gone through the owner. This is done by tracking an extra flag in DynamoDB.

## How was this patch tested?

Existing tests

## Does this PR introduce _any_ user-facing changes?

Yes, introduces new configs (see DeltaSQLConf changes) which can be used to configure the DynamoDBCommitOwner.

### CODE REVIEW

File: spark/src/main/java/io/delta/dynamodbcommitstore/DynamoDBCommitOwnerClient.java [hunk 1]
```diff
@@ -655,16 +709,27 @@ private void tryEnsureTableExists() throws IOException {
                 }
             }
             if (status.equals("ACTIVE")) {
+                if (created) {
+                    LOG.info("Successfully created DynamoDB table `{}`", managedCommitsTableName);
```
Comment (prakharjain09) [inline spark/src/main/java/io/delta/dynamodbcommitstore/DynamoDBCommitOwnerClient.java:713]: Could we add couple of log statements in Commit method as well.

- Attempting to commit <> via DynamoDB
- Commit <> done successfully. Backfilling commit files.

File: spark/src/main/java/io/delta/dynamodbcommitstore/DynamoDBCommitOwnerClientBuilder.java [hunk 2]
```diff
@@ -62,27 +54,40 @@ public CommitOwnerClient build(SparkSession spark, Map<String, String> conf) {
...
-            InvocationTargetException,
-            InstantiationException,
-            IllegalAccessException {
-        Class<?> awsCredentialsProviderClass = Class.forName(credentialProviderName);
+            String credentialProviderName,
+            Configuration hadoopConf
+    ) throws ReflectiveOperationException {
         AWSCredentialsProvider awsCredentialsProvider =
-                (AWSCredentialsProvider) awsCredentialsProviderClass.getConstructor().newInstance();
+                ReflectionUtils.createAwsCredentialsProvider(credentialProviderName, hadoopConf);
```
Comment (prakharjain09) [inline spark/src/main/java/io/delta/dynamodbcommitstore/DynamoDBCommitOwnerClientBuilder.java:90]: Is this similar to how DynamoDBLogStore do it?

File: spark/src/test/scala/org/apache/spark/sql/delta/managedcommit/CommitOwnerClientImplSuiteBase.scala [hunk 3]
```diff
@@ -204,13 +204,7 @@ trait CommitOwnerClientImplSuiteBase extends QueryTest
...
       assert(e.getMessage === "Commit version 0 must go via filesystem.")
       writeCommitZero(logPath)
-      var expectedVersion = -1
-      if (tableCommitOwnerClient.commitOwnerClient.isInstanceOf[DynamoDBCommitOwnerClient]) {
-        // DynamoDBCommitOwnerClient stores attemptVersion as the current table version when
-        // registerTable is called.
-        expectedVersion = 0
-      }
-      assert(tableCommitOwnerClient.getCommits() == GetCommitsResponse(Seq.empty, expectedVersion))
+      assert(tableCommitOwnerClient.getCommits() == GetCommitsResponse(Seq.empty, -1))
```
Comment (prakharjain09) [inline spark/src/test/scala/org/apache/spark/sql/delta/managedcommit/CommitOwnerClientImplSuiteBase.scala:207]: Could we add test to validate that the spark config overrides are correctly used by DynamoDBCommitOwnerClient.

File: spark/src/main/java/io/delta/dynamodbcommitstore/DynamoDBCommitOwnerClientBuilder.java (hunk 2 above)
Comment (dhruvarya-db) [reply spark/src/main/java/io/delta/dynamodbcommitstore/DynamoDBCommitOwnerClientBuilder.java:90]: Yes, this has been directly copied from there.

File: spark/src/test/scala/org/apache/spark/sql/delta/managedcommit/CommitOwnerClientImplSuiteBase.scala (hunk 3 above)
Comment (dhruvarya-db) [reply spark/src/test/scala/org/apache/spark/sql/delta/managedcommit/CommitOwnerClientImplSuiteBase.scala:207]: Added a test for this: https://github.com/delta-io/delta/pull/3130/files#diff-dc7d3ff136c9f5338c9524783e00375fc251c212230e0b366581b1e5b53b8387R242

**Comment by felipepessoto** [comment]
@dhruvarya-db any plans to add support to Azure (CosmosDB?) commit owner?

**Review by prakharjain09 — APPROVED** [review]

**Comment by prakharjain09** [comment]
> @dhruvarya-db any plans to add support to Azure (CosmosDB?) commit owner?

@felipepessoto There is no specific plan right now. Currently the API surface is still evolving. Once it stablizes (probably by Delta 4.0), anyone should be able to implement the API and define their own commit owner.

We chose DynamoDB for reference implementation as S3 doesn't have putIfAbsent which causes lost writes issue as explained [here](https://github.com/delta-io/delta/issues/39).

**Pull request merged**